src/psyclone/tests/psyir/nodes/loop_test.py

```diff
@@ -190,7 +190,8 @@ def test_loop_create():
     start = Literal("0", DataType.INTEGER)
     stop = Literal("1", DataType.INTEGER)
     step = Literal("1", DataType.INTEGER)
-    child_node = Assignment.create(Reference("tmp"), Reference("i"))
+    child_node = Assignment.create(Reference(DataSymbol("tmp", DataType.REAL)),
+                                   Reference(DataSymbol("i", DataType.REAL)))
     loop = Loop.create("i", start, stop, step, [child_node])
```

Comment:
Something for another issue but `Loop.create()` really ought to take a symbol for the loop variable, rather than just a name?

Reply:
Yes agreed. Created issue #679